src/test/java/picard/fingerprint/CrosscheckReadGroupFingerprintsTest.java

```diff
@@ -0,0 +1,312 @@
+        doTest(args, metrics, expectedRetVal, expectedNMetrics, expectedType, false);
+    }
+
+    private void doTest(final String[] args, final File metrics, final int expectedRetVal, final int expectedNMetrics, final CrosscheckMetric.DataType expectedType, final boolean expectAllMatch) throws IOException {
+
+        final CrosscheckReadGroupFingerprints crossChecker = new CrosscheckReadGroupFingerprints();
+        Assert.assertEquals(crossChecker.instanceMain(args), expectedRetVal);
+
+        final MetricsFile<CrosscheckMetric, Comparable<?>> metricsOutput = new MetricsFile<>();
+        metricsOutput.read(new FileReader(metrics));
+
+        Assert.assertFalse(metricsOutput.getMetrics().stream()
+                .anyMatch(m -> m.DATA_TYPE != expectedType));
+
+        Assert.assertFalse(metricsOutput.getMetrics().stream()
+                .anyMatch(m-> m.LOD_SCORE_NORMAL_TUMOR == null));
+        Assert.assertFalse(metricsOutput.getMetrics().stream()
+                .anyMatch(m-> m.LOD_SCORE == null));
+        Assert.assertFalse(metricsOutput.getMetrics().stream()
+                .anyMatch(m-> m.LOD_SCORE_TUMOR_NORMAL == null));
+
+        if (expectAllMatch) {
+            Assert.assertTrue(metricsOutput.getMetrics().stream()
+                    .allMatch(m -> m.RESULT == CrosscheckMetric.FingerprintResult.INCONCLUSIVE ||
+                            m.RESULT.isMatch() == m.LEFT_SAMPLE.equals(m.RIGHT_SAMPLE)));
+        } else if (expectedRetVal == 0) {
+            Assert.assertTrue(metricsOutput.getMetrics().stream()
+                    .allMatch(m -> m.RESULT == CrosscheckMetric.FingerprintResult.INCONCLUSIVE ||
+                            m.RESULT.isExpected()));
+        } else {
+            Assert.assertTrue(metricsOutput.getMetrics().stream()
+                    .anyMatch(m -> !m.RESULT.isExpected()));
+        }
+
+        Assert.assertEquals(metricsOutput.getMetrics().size(), expectedNMetrics);
+
+        // at the readgroup level things aren't always conclusive...
+        if (!metricsOutput.getMetrics().isEmpty() && expectedType != CrosscheckMetric.DataType.READGROUP) {
+            Assert.assertTrue(metricsOutput.getMetrics().stream()
+                    .anyMatch(m -> m.RESULT != CrosscheckMetric.FingerprintResult.INCONCLUSIVE));
+        }
+
+        //check that fields that should have an actual value, indeed do
+        for(final String fieldName : lookupMap.get(expectedType)) {
+            try {
+                final Field field = CrosscheckMetric.class.getField(fieldName);
+                Assert.assertTrue(metricsOutput.getMetrics().stream().allMatch(m -> {
+                    try {
+                        return field.get(m) != multipleValuesString && field.get(m) != null;
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+                        return false;
```

Comment:
Since we know this is going to hit the assert above and throw, why not let the exception get thrown in this case? That would avoid the need to print the stack trace here. Same comment applies to line 240 below.

Reply:
According to https://stackoverflow.com/questions/41177031/throwing-exception-inside-the-java-8-stream-foreach this cannot be done since the signature for the function in "allMatch" doesn't throw...

Reply:
You could catch & rethrow a `RuntimeException` to handle that. But it looks like what you want to do here is fail as soon as any metric value is not correct. E.g.
```
for (final String fieldName: lookupMap.get(expectedType)) {
    final Field field = CrosscheckMetric.class.getField(fieldName);
    for (CrosscheckMetric metric : metricsOutput.getMetrics()) {
        Assert.assertNotNull(field.get(metric));
        Assert.assertNotEquals(field.get(metric), multipleValuesString);
    }
}
```
This will give you more precise information about the failure when it occurs as well.